### PR TITLE
[feat] improve type safety for Named, servantify brig internal route

### PR DESCRIPTION
**Pull request by MangoIV**
https://wearezeta.atlassian.net/browse/WPB-1224
- improve type safety of Named by making it possible to rule out weakly typed arguments to the type (e.g. Type)
- servantify the internal route for querying the teams API for servant

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: libs/wire-api/src/Wire/API/Routes/Internal/Brig.hs [hunk 1]
```diff
@@ -664,6 +669,81 @@ type TeamsAPI =
...
+    "suspend-team"
+    ( "teams"
+        :> Capture "tid" TeamId
+        :> "suspend"
+        :> Post
+             '[Servant.JSON]
+             NoContent
+    )
+
+type UnSuspendTeam =
```
Comment (elland) [inline libs/wire-api/src/Wire/API/Routes/Internal/Brig.hs:718]: `UnsuspendTeam` makes more sense to me. The current spelling makes it look like a typical Haskell unwrap of newtypes.

File: libs/wire-api/src/Wire/API/Routes/Named.hs [hunk 2]
```diff
@@ -33,7 +33,9 @@ import Servant.Client.Core (clientIn)
 import Servant.OpenApi
 
 -- | See http://docs.wire.com/developer/developer/servant.html#named-and-internal-route-ids-in-swagger
-newtype Named name x = Named {unnamed :: x}
+type Named name = UntypedNamed (IsStronglyTyped name)
+
+newtype UntypedNamed name x = Named {unnamed :: x}
```
Comment (elland) [inline libs/wire-api/src/Wire/API/Routes/Named.hs:38]: I think some comments explaining what and why here might be good for posterity :)

**Review by elland — COMMENTED** [review]
How do you feel about moving the api handler definitions to the same file where we define the routes?

File: libs/wire-api/src/Wire/API/Routes/Internal/Brig.hs (hunk 1 above)
Comment (MangoIV) [reply libs/wire-api/src/Wire/API/Routes/Internal/Brig.hs:718]: haha, you're right, I will change it :D

**Comment by MangoIV** [comment]
> How do you feel about moving the api handler definitions to the same file where we define the routes?

@elland the file I would move to is already quite large and the `Brig.Team.API` file contains the public routes as well, so that was my reason to not move it, if you wish though, I will move it over. 

Maybe also consider moving the API definition itself to `Brig.Team.API`, similarly how it is done with the [exported `servantAPI`](https://github.com/wireapp/wire-server/blob/c15986afbdbf38db81ed67fad465cc98d882ffd3/services/brig/src/Brig/Team/API.hs#L81)

File: libs/wire-api/src/Wire/API/Routes/Internal/Brig.hs [hunk 3]
```diff
@@ -664,6 +669,81 @@ type TeamsAPI =
...
+        :> Get '[Servant.JSON] FoundInvitationCode
+    )
+
+newtype FoundInvitationCode = FoundInvitationCode User.InvitationCode
+  deriving stock (Eq, Show, Generic)
+  -- TODO: is this correct?
+  deriving newtype (S.ToSchema)
+
+instance ToJSON FoundInvitationCode where
+  toJSON (FoundInvitationCode c) = Aeson.object ["code" Aeson..= c]
```
Comment (MangoIV) [inline libs/wire-api/src/Wire/API/Routes/Internal/Brig.hs:705]: I also would like to ask if this is correct, especially the `S.ToSchema` part which does not adhere to the `ToJSON` instance but I don't know how / if they should be related.

**Review by elland — APPROVED** [review]

**Comment by MangoIV** [comment]
*finally the tests have passed!*

**Pull request merged**